Buttplug/Server/Bluetooth/Devices/Cueme.cs

```diff
@@ -0,0 +1,213 @@
+        }
+    }
+
+    internal class Cueme : ButtplugBluetoothDevice
+    {
+        internal struct CuemeType
+        {
+            public string Name;
+            public uint VibeCount;
+        }
+
+        private static readonly Dictionary<uint, CuemeType> DevInfos = new Dictionary<uint, CuemeType>
+        {
+            { 1, new CuemeType() { Name = "Mens", VibeCount = 8 } },
+            { 2, new CuemeType() { Name = "Bra", VibeCount = 8 } },
+            { 3, new CuemeType() { Name = "Womens", VibeCount = 4 } },
+        };
+
+        private readonly CuemeType _devInfo;
+        private double[] _vibratorSpeeds = { 0, 0, 0, 0, 0, 0, 0, 0 };
+        internal static readonly double[] NullSpeed = { 0, 0, 0, 0, 0, 0, 0, 0 };
+        private uint _vibeIndex;
+
+        // When alternating vibes, switch twice a second
+        internal static readonly uint DelayTimeMS = 500;
+
+        private readonly System.Timers.Timer _updateValueTimer = new System.Timers.Timer();
+        private CancellationTokenSource _stopUpdateCommandSource = new CancellationTokenSource();
+
+        public Cueme(IButtplugLogManager aLogManager,
+                       IBluetoothDeviceInterface aInterface,
+                       IBluetoothDeviceInfo aInfo)
+            : base(aLogManager,
+                   "Cueme Unknown",
+                   aInterface,
+                   aInfo)
+        {
+            var bits = aInterface.Name.Split('_');
+            if (bits.Length == 3 && uint.TryParse(bits[2], out var typeNum) && DevInfos.ContainsKey(typeNum))
+            {
+                _devInfo = DevInfos[typeNum];
+            }
+            else
+            {
+                BpLogger.Warn($"Cannot identify Cueme device {Name}, defaulting to Womens settings.");
+                _devInfo = DevInfos[3];
+            }
+
+            Name = $"Cueme {_devInfo.Name}";
+
+            // Create a new timer that wont fire any events just yet
+            _updateValueTimer.Interval = DelayTimeMS;
+            _updateValueTimer.Elapsed += CuemeUpdateHandler;
+            _updateValueTimer.Enabled = false;
+            aInterface.DeviceRemoved += OnDeviceRemoved;
+
+            AddMessageHandler<StopDeviceCmd>(HandleStopDeviceCmd);
+            AddMessageHandler<SingleMotorVibrateCmd>(HandleSingleMotorVibrateCmd);
+            AddMessageHandler<VibrateCmd>(HandleVibrateCmd, new MessageAttributes { FeatureCount = _devInfo.VibeCount });
+        }
+
+        private void OnDeviceRemoved(object aEvent, EventArgs aArgs)
+        {
+            // Timer should be turned off on removal.
+            _updateValueTimer.Enabled = false;
+
+            // Clean up event handler for that magic day when devices manage to disconnect.
+            Interface.DeviceRemoved -= OnDeviceRemoved;
+        }
+
+        private async void CuemeUpdateHandler(object aEvent, ElapsedEventArgs aArgs)
+        {
+            if (aArgs == null && !_updateValueTimer.Enabled)
```

Comment:
Call this handler directly then start the timer where you need to. Do not treat the elapsed args being null as a state variable. That's not how event handlers should be treated.